### ISSUE TITLE
Misleading diagnostic when `Clone::clone` is called on an unclonable type

### ISSUE THREAD

**Issue by 0x7CFE** [rust-lang/rust#86146]
Given the following code: 

```rust
struct Unclonable;

fn main() {
    let original = vec![vec![Unclonable]];
    let clone = original.clone();
}
```

The current output is:

```
error[E0599]: the method `clone` exists for struct `Vec<Vec<Unclonable>>`, but its trait bounds were not satisfied
   --> src/main.rs:7:26
    |
5   |       let clone = original.clone();
    |                            ^^^^^ method cannot be called on `Vec<Vec<Unclonable>>` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `Vec<Unclonable>: Clone`
            which is required by `Vec<Vec<Unclonable>>: Clone`
```

Current diagnostic is a bit misleading, since the issue is not with vector per se, but with a struct. Ideally, the diagnostic should dive into the type and give a hint to the actual cause. This is especially strange since a minor change to the source results in a much better error output:

```diff
-    let original = vec![vec![Unclonable]];
+    let original = vec![Unclonable];
```

```
error[E0599]: the method `clone` exists for struct `Vec<Unclonable>`, but its trait bounds were not satisfied
   --> src/main.rs:7:26
    |
1   |   struct Unclonable;
    |   ------------------ doesn't satisfy `Unclonable: Clone`
...
5   |       let clone = original.clone();
    |                            ^^^^^ method cannot be called on `Vec<Unclonable>` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `Unclonable: Clone`
            which is required by `Vec<Unclonable>: Clone`
```

**Comment by estebank** [comment]
Triage: little change

```
error[E0599]: the method `clone` exists for struct `Vec<Vec<Unclonable>>`, but its trait bounds were not satisfied
 --> src/main.rs:5:26
  |
5 |     let clone = original.clone();
  |                          ^^^^^ method cannot be called on `Vec<Vec<Unclonable>>` due to unsatisfied trait bounds
 --> /rustc/92c1937a90e5b6f20fa6e87016d6869da363972e/library/alloc/src/vec/mod.rs:400:1
  |
  = note: doesn't satisfy `Vec<Unclonable>: Clone`
  |
  = note: doesn't satisfy `Vec<Vec<Unclonable>>: Clone`
  |
  = note: the following trait bounds were not satisfied:
          `Vec<Unclonable>: Clone`
          which is required by `Vec<Vec<Unclonable>>: Clone`
```

**Comment by estebank** [comment]
Current output:

```
error[E0599]: the method `clone` exists for struct `Vec<Vec<Unclonable>>`, but its trait bounds were not satisfied
   --> src/main.rs:5:26
    |
5   |     let clone = original.clone();
    |                          ^^^^^ method cannot be called on `Vec<Vec<Unclonable>>` due to unsatisfied trait bounds
    |
   ::: /playground/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:397:1
    |
397 | pub struct Vec<T, #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global> {
    | ------------------------------------------------------------------------------------------------ doesn't satisfy `Vec<Unclonable>: Clone` or `Vec<Vec<Unclonable>>: Clone`
    |
    = note: the following trait bounds were not satisfied:
            `Vec<Unclonable>: Clone`
            which is required by `Vec<Vec<Unclonable>>: Clone`
```
